### PR TITLE
chore/1596:  add value to input elements sandbox code snippet

### PR DESCRIPTION
**Pull request by vanessatran-ddi**
Add `value` to input elements's sandbox. 

**Dropdown:** 
Before: 
image

After:
image

**Form Item:**
Before: 
image

After: 
image

**Input:**
Before: 
image
After: 
image

**Textarea:**
Before: 
image

After: 
image

### CODE REVIEW

**Comment by syedszeeshan** [comment]
@vanessatran-ddi That's great, I do have a solution for the my issue#1586 for the generated code and I have created some test scenarios in angular playground for various combination. Will confirm with Dustin and will also share with you after.

File: src/routes/components/Checkbox.tsx [hunk 1]
```diff
@@ -29,15 +29,15 @@ export default function CheckboxPage() {
   const [checkboxProps, setCheckboxProps] = useState<ComponentPropsType>({
     name: "item",
     checked: false,
-    text: "Item",
-    value: "",
+    text: "Item"
```
Comment (chrisolsen) [inline src/routes/components/Checkbox.tsx:32]: Keep commas at the end. This should be done by the prettier formatter.

File: src/routes/components/Checkbox.tsx [hunk 2]
```diff
@@ -126,10 +126,10 @@ export default function CheckboxPage() {
                 allowCopy={true}
                 code={`
                   export class SomeOtherComponent {
-                    checked = false;
+                    theChecked: boolean = false;
```
Comment (chrisolsen) [inline src/routes/components/Checkbox.tsx:129]: Why is the variable name being changed?

File: src/routes/components/TextArea.tsx [hunk 3]
```diff
@@ -191,7 +200,7 @@ export default function TextAreaPage() {
       type: "(name: string, value: string[] | string | null) => void",
       required: true,
       description: "Callback function when textarea value is changed",
-    },
+    }
```
Comment (chrisolsen) [inline src/routes/components/TextArea.tsx:203]: prettier

File: src/routes/components/TextArea.tsx [hunk 4]
```diff
@@ -248,16 +258,16 @@ export default function TextAreaPage() {
...
                 onChange(name: string, value: string) {
-                  setValue(value);
+                  setTheValue(value);
                 }
             `}
               />
 
               <GoAFormItem {...formItemProps}>
-                <GoATextarea {...componentProps} value="" onChange={noop} />
+                <GoATextarea {...componentProps} width="60ch" name="item" value="" onChange={noop} />
```
Comment (chrisolsen) [inline src/routes/components/TextArea.tsx:270]: Why is width being set here?

File: src/routes/components/TextField.tsx [hunk 5]
```diff
@@ -41,7 +41,7 @@ type CastingType = {
 export default function TextFieldPage() {
   const [componentProps, setComponentProps] = useState<ComponentPropsType>({
     name: "item",
-    value: "",
+    width: "20ch",
```
Comment (chrisolsen) [inline src/routes/components/TextField.tsx:44]: Why is this width being set?

File: src/routes/components/TextField.tsx [hunk 6]
```diff
@@ -415,6 +424,7 @@ export default function TextFieldPage() {
                 code={`
                 // non-reactive code
                 export class SomeComponent {
+                  theValue: string = "";
```
Comment (chrisolsen) [inline src/routes/components/TextField.tsx:427]: The variable renaming seems unnecessary

File: src/routes/components/TextField.tsx [hunk 7]
```diff
@@ -438,16 +448,16 @@ export default function TextFieldPage() {
...
                 onChange(name: string, value: string) {
-                  setValue(value);
+                  setTheValue(value);
                 }
               `}
               />
 
               <GoAFormItem {...formItemProps}>
-                <GoAInput {...componentProps} onChange={noop} />
+                <GoAInput {...componentProps} name="item" value="" onChange={onChange}/>
```
Comment (chrisolsen) [inline src/routes/components/TextField.tsx:460]: In some places the useState is removed, but it is modified here. Is there a reason for that?

File: src/components/sandbox/AngularSerializer.ts [hunk 8]
```diff
@@ -21,7 +21,7 @@ export class AngularSerializer extends BaseSerializer implements Serializer {
   }
 
   #dynamicProp(name: string): string {
-    return `[${name.toLowerCase()}]="some${this.capitalize(name)}Value"`;
+    return `[${name.toLowerCase()}]="the${this.capitalize(name)}"`;
```
Comment (chrisolsen) [inline src/components/sandbox/AngularSerializer.ts:24]: Does this have to be changed?

**Comment by lizhuomeng71** [comment]
Dropdown Compoment, Second Code Snipt, FormControl, has the following issue

itemFormCtrl should be renamted to colorsFormCtrl
disabled=true error=true filterable=true  native=true should be disabled="true" error="true"  filterable="true"  native="true"
Screenshot 2024-04-25 131528
Screenshot 2024-04-25 131516
Screenshot 2024-04-25 131450

**Comment by lizhuomeng71** [comment]
Form Item Compoment, Second Code Snipt, FormControl, has the following issue

itemFormCtrl should be renamted to firstNameFormCtrl

image
image

**Comment by lizhuomeng71** [comment]
Iput Compoment, Second Code Snipt, FormControl, has the following issue

disabled=true error=true  should be disabled="true" error="true" 

image

**Comment by lizhuomeng71** [comment]
textarea Compoment, Second Code Snipt, FormControl, has the following issue

disabled=true error=true showcount=true  should be disabled="true" error="true" showcount="true"

**Comment by lizhuomeng71** [comment]
Hi @vanessatran-ddi , the Type 'string' is not assignable to type 'boolean'. still exist for dropdown and input

**Comment by lizhuomeng71** [comment]
Issue is verified on chrome, edge and firefox

**Review by chrisolsen — APPROVED** [review]

**Pull request merged**